src/components/transaction.js

```diff
@@ -1,7 +1,72 @@
-import React from "react";
+import React, { useState } from "react";
+import { Navbar } from "./common-session-component/navbar";
+import { AccountBalance } from "./common-session-component/acct-balance";
+import {
+  getAllUsers,
+  getUserByAccountNumber,
+  getUserIndexByAccountNumber,
+} from "../helpers/user-helper";
 
 export const Transaction = () => {
-  return <h1>Transaction page</h1>;
-};
+  const currentUserAccountNumber = JSON.parse(
+    localStorage.getItem("MB_LOGGEDIN_USER_ACCOUNT_NUMBER")
+  );
+
+  const currentUser = getUserByAccountNumber(currentUserAccountNumber);
+  const userTransactions = currentUser.transactions;
+  const [userTransactionArr, setUserTransactionArr] =
+    useState(userTransactions);
 
-export default Transaction;
+  const clearTransactionHistory = () => {
+    const registeredUsers = getAllUsers();
+    const currentUserIndex = getUserIndexByAccountNumber(
+      currentUserAccountNumber
+    );
+    registeredUsers[currentUserIndex].transactions = [];
+    localStorage.setItem("MB_USER_ACCOUNTS", JSON.stringify(registeredUsers));
+    setUserTransactionArr([]);
+  };
+
+  return (
+    <div>
+      <Navbar />
+      <AccountBalance />
+      <h1>Transaction History</h1>;
+      <button
+        className="btn-danger"
+        type="button"
+        onClick={() => clearTransactionHistory()}
+      >
+        Clear transaction history
+      </button>
+      <div>
+        <table className=" container-fixed">
+          <thead>
+            <tr>
+              <th>Timestamp</th>
+              <th>Transaction Reference</th>
+              <th>Type</th>
+              <th>Amount</th>
+              <th>Balance Before</th>
+              <th>Balance After</th>
+              <th>Beneficiary</th>
+            </tr>
+          </thead>
+          <tbody>
+            {userTransactionArr.map((transaction) => (
+              <tr>
```

Comment:
Add key prop with transaction reference as value, as it's the only unique value of transaction.
`key={transaction.transactionReference}`